Plugins/Autodiff/CMakeLists.txt

```diff
@@ -1,24 +1,25 @@
-include(ExternalProject)
-
 add_library(
-  ActsAutodiffPlugin INTERFACE)
-
-add_dependencies(
-  ActsAutodiffPlugin autodiff)
+  ActsPluginAutodiff INTERFACE)
 
 target_include_directories(
-  ActsAutodiffPlugin
+  ActsPluginAutodiff
   INTERFACE
-  $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/thirdparty/autodiff>
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/>
+    $<INSTALL_INTERFACE:include>)
+ 
+# the target name is different when using find_package(...) or add_subdirectory(...)
+if(ACTS_USE_SYSTEM_AUTODIFF)
+    set(AUTODIFF_TARGET autodiff::autodiff)
+else()
+    set(AUTODIFF_TARGET autodiff)
+endif()
```

Comment:
Suggestion: add the missing `autodiff::autodiff` target as an alias target via `add_library(<name> ALIAS <target>)` when you include it via `add_subdirectory` in the main `CMakeLists.txt`. You can then always use the `autodiff::autodiff` target regardless of how the package is included. The plugin should not care whether we use the bundled version or not. It also means we have to do this only in one location if we e.g. decide to use the autodiff library somwhere else.

In the long term, we should try to get CMake fixes upstream so we do not need to patch this locally. But not for this PR.